message/pool/message.go

```diff
@@ -75,6 +76,9 @@ func (r *Message) ResetOptionsTo(in message.Options) {
 		r.valueBuffer = append(r.valueBuffer, make([]byte, used)...)
 		opts, used, err = r.msg.Options.ResetOptionsTo(r.valueBuffer, in)
 	}
+	if err != nil {
+		panic(fmt.Errorf("unexpected error: %w", err))
```

Comment:
@jkralik panicking in these setters feels wrong. But what else can we do without changing the API? If the first call `r.msg.Options.ResetOptionsTo` gives us correct required size for the value buffer this should never occur. But any ideas how else might we enforce it?